### ISSUE TITLE
Trouble Using mkdocstrings In Conjunction with mkdocs-gen-files

### ISSUE THREAD

**Issue by samgaudet** [mkdocstrings/mkdocstrings#399]
# TL;DR

While trying to use `mkdocstrings` alongside `mkdocs-gen-files`, I am getting a vague error that I am having trouble understanding:

```
> mkdocs serve

ERROR    -  mkdocstrings: No module named ' '
ERROR    -  Error reading page 'reference\__init__.md':
ERROR    -  Could not collect ' '
```

# Details

First off, thank you for all the work on this project&mdash;`mkdocstrings` is awesome, and I'm looking forward to finishing off getting this automation working. I have been following the guide [here](https://mkdocstrings.github.io/recipes/) on automating the generation of our reference files, and eventually the automation of the nav being generated, etc.

I have tried searching the errors above to no avail&mdash;any help or tips about what to check next are greatly appreciated. Thanks!

# Environment Info

Here are the package versions for relevant `mkdocs` plugins, etc.:

```
mkdocs                     1.2.3
mkdocs-autorefs            0.3.1
mkdocs-gen-files           0.3.4
mkdocs-literate-nav        0.4.1
mkdocs-material            8.2.4
mkdocs-material-extensions 1.0.3
mkdocs-section-index       0.3.3
mkdocstrings               0.18.1
mkdocstrings-python        0.6.6
mkdocstrings-python-legacy 0.2.2
```

# Folder structure

Skipped a few files but I believe this describes the general shape of things:

```
.
├── docs
│   ├── generate_reference_pages.py
│   ├── explanation
│   │      └── information.md
│   ├── how_to_guides
│   │      ├── auth.md
│   │      └── sheets.md
│   ├── images
│   ├── tutorials
│   ├── features.md
│   ├── goals.md
│   ├── index.md
│   ├── requirements.txt
│   └── SUMMARY.md  # intending to use later for auto-nav
├── docs_src
│   ├── conf.py
│   └── docs.rst
├── examples
│   ├── sample.py
│   └── <lots of other example code>
├── pygsuite
│   ├── __init__.py
│   └── <other python modules in package>
├── tests
│   ├── conftest.py
│   └── <lots of other testing code>
├── mkdocs.yml
├── pyproject.toml
├── pytest.ini
├── readthedocs.yml
├── requirements.txt
└── setup.py
```

# Mkdocs configs (`mkdocs.yml`)

My `mkdocs.yml` file contains a `nav` section, referencing the locations of the docs that are to-be created such as:

```yaml
nav:
  - Introduction:
    - Overview: index.md
    - Goals: goals.md
    - Features: features.md
. . .
  - Reference Guides:
    - Auth: reference/auth/authorization.md
    - Drive:
      - DriveObject: reference/drive/drive_object.md
      - File: reference/drive/file.md
      - Folder: reference/drive/folder.md
```

as well as the relevant plugin information:

```yaml
plugins:
  - gen-files:
      scripts:
        - docs/generate_reference_pages.py
  - mkdocstrings:
      watch:
        - pygsuite
```

**Comment by samgaudet** [comment]
Should have added that this is the content of `docs/generate_reference_pages.py`:

```py
"""Generate the code reference pages."""

from pathlib import Path

import mkdocs_gen_files

# nav = mkdocs_gen_files.Nav()

for path in sorted(Path("pygsuite").rglob("*.py")):
    module_path = path.relative_to("pygsuite").with_suffix("")
    doc_path = path.relative_to("pygsuite").with_suffix(".md")
    full_doc_path = Path("reference", doc_path)

    parts = list(module_path.parts)

    if parts[-1] == "__init__":
        parts = parts[:-1]
        # doc_path = doc_path.with_name("index.md")
        # full_doc_path = full_doc_path.with_name("index.md")
    elif parts[-1] == "__main__":
        continue

    # nav[parts] = str(doc_path)

    with mkdocs_gen_files.open(full_doc_path, "w") as fd:
        identifier = ".".join(parts)
        print("::: " + identifier, file=fd)

    mkdocs_gen_files.set_edit_path(full_doc_path, path)

# with mkdocs_gen_files.open("reference/SUMMARY.md", "w") as nav_file:
#     nav_file.writelines(nav.build_literate_nav())

```

**Comment by samgaudet** [comment]
Ah! Is it possible this issue arises from having all of these installed simultaneously?

```
mkdocstrings               0.18.1
mkdocstrings-python        0.6.6
mkdocstrings-python-legacy 0.2.2
```

**Comment by pawamoy** [comment]
Hint: `mkdocstrings: No module named ' '`, you have written an empty identifier to one of the auto-generated pages :slightly_smiling_face: 

It comes from this:

```python
module_path = path.relative_to("pygsuite").with_suffix("")
```

For the `__init__.py` module, it gives `Path(__init__)`. Then

```python
    if parts[-1] == "__init__":
        parts = parts[:-1]
```

...and the parts become `[]`.

Solution: don't compute the path relative to `pygsuite`. Only do this for `src` layouts, to get rid of the `src` part (you don't want to get rid of the `pygsuite` part :wink: )

**Comment by samgaudet** [comment]
D'oh&mdash;at least it was a simple issue, thanks so much for the help!

**Comment by pawamoy** [comment]
It seems to be the same reason as @samgaudet:

- with `src = Path(__file__).parent / "quantfreedom"`, `src` points at your package
- with `module_path = path.relative_to(src).with_suffix("")`, you strip the package name from the path
- `parts = parts[:-1]` then gives `[]`

Possible solution: switch to an actual src-layout (put `quantfreedom` into an `src` folder), or use `relative_to(src.parent)` to prevent stripping the package name.
